adagrams/game.py

```diff
@@ -1,11 +1,142 @@
+    return selection
 
-def uses_available_letters(word, letter_bank):
-    pass
 
+    # pass
+draw_letters()
+def uses_available_letters(word, letter_bank):
+    """
+    Turn all letters to uppercases if they are not already uppercase
+    """
+    if not word.isupper():
+        word = word.upper()
+    """
+    make a copy of the letter_bank list
+    remove every letter of the word from the bank
+    count the number of letters that are in the bank
+    """
+    counter = 0
+    bank = letter_bank.copy()
+    for letter in word:
+        if letter in bank:
+            counter += 1
+            bank.remove(letter)
+    """
+    return True when all of the letters of the word was in the bank
+    """
+    return counter == len(word)
+    # pass
+uses_available_letters("alex", ["A","B","C","D"])
 def score_word(word):
-    pass
+
+    """
+    return 0 if the word is empty
+    """
+    if not word:
+        return 0
+    
+    word=word.upper()
+    """
+    add 7 scores if the lenght is between 7 and 10
+    """
+    score = 0
+    if 7 <= len(word) <= 10:
+        score += 8
+
+    my_dic= { "A":1, "E":1, "I" :1, "O":1, "U" :1, "L" :1, "N" :1, "R":1, "S":1, "T" : 1,
+    "D":2, "G":2,
+    "B":3, "C":3, "M":3, "P":3,
+    "F":4, "H":4, "V":4, "W":4, "Y":4,
+    "K":5,
+    "J":8, "X":8,
+    "Q":10, "Z": 10}
+    """
+    loop over the word elements, find the element key in my_dic dictionary \ 
+    and add the related score (related value) to the result
+    """
+    for element in word:
+        if element in my_dic:
+            score += my_dic[element]
+    return score
+    """
+    Following is an alternative Approach
+    """
+    # if not word.isupper():
+    #     word = word.upper()
+    # score_one = {1:["A","E","I","O","U","L","N","R","S","T"],2:["D","G"],3:["B", "C", "M", "P"],4:["F", "H", "V", "W", "Y"],5: ["k"], 8:["J","X"], 10:["Q","Z"]}
+    # total_score = 0
+    # for letter in word:
+    #     for score, letters in score_one.items(): 
+    #         if letter in letters:
+    #             total_score += score
+    # if 7 <= len(word) <= 10:
+    #     total_score += 8
+    
+    # return total_score
+
+
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+
+    counter=0
+    """
+    loop over the word_list, calculate the score by calling score_word function
+    compare the score with the one that is stored in counter and update the counter variable if 
+    the new score is higher than the previous one
+    elif another word with the same highest score was found apply 
+    further criteria to select the result
+    """
+    for word in word_list:
+        word_score = score_word(word)
+        if word_score > counter:
+            res = (word, word_score)
+            counter = word_score
+        elif word_score == counter and len(res[0]) != 10:
+        
+            if len(word) == 10 or len(word) < len(res[0]):
+                res = (word, word_score)
+            
+    return res
```

Comment:
Nice! This is a really clean solution. A few suggestions:

- Is `res` short for `result` ? It's not mentioned in your doc strings either. Let's use more descriptive name so we don't have to make assumptions about what the data is supposed to represent. 
- No need for the empty space on line 123.